### ISSUE TITLE
String.format ignores case on placeholder

### ISSUE THREAD

**Issue by AlexHolly** [godotengine/godot#20428]
https://github.com/godotengine/godot/commit/f8e8ac2c664a12f3af2b9bb545da193d86c75503

Edit: replace `lname` with `uname`
```
var rs = "Hi, {name} ... Uname}!!!? uname} ;( ".format([["name","Godette"]])
rs = rs.format([["name","Godette".to_upper()]], "U_}")
print(rs.format([["name","Godette".to_lower()]], "u_}"))
```
Should print
`Hi, Godette ... GODETTE!!!? godette ;(`

but shows
`Hi, Godette ... GODETTE!!!? GODETTE ;(`

Easy to fix, use "replace" instead of "replacen" 
https://github.com/godotengine/godot/blob/master/core/ustring.cpp#L2804
https://github.com/godotengine/godot/blob/master/core/ustring.cpp#L2784
https://github.com/godotengine/godot/blob/master/core/ustring.cpp#L2771

Also need to have a look at the last comments in https://github.com/godotengine/godot/issues/5493

**Comment by akien-mga** [comment]
There are issues in your code snippet, it doesn't run.

**Comment by akien-mga** [comment]
I guess it was meant to be:
```gdscript
	var rs = "Hi, {name} ... Uname}!!!? uname} ;( ".format([["name","Godette"]])
	rs = rs.format([["name","Godette".to_upper()]], "U_}")
	print(rs.format([["name","Godette".to_lower()]], "u_}"))
```
Man is this string formatting convoluted and unreadable...

**Comment by akien-mga** [comment]
Since you implemented it (#7010), you're probably the best equipped dev to fix it :)
